### ISSUE TITLE
rejected shares on yescryptr8g at zergpool

### ISSUE THREAD

**Issue by Quake4** [JayDDee/cpuminer-opt#261]
![image](https://user-images.githubusercontent.com/29517243/79376017-36bb7980-7f62-11ea-91a3-321aba86dd68.png)

**Comment by JayDDee** [comment]
Reproduced. Works in v3.12.7. Under investigation.

**Comment by JayDDee** [comment]
I found the code that broke it but don't know why.

In v3.12.8 I eliminated a function in the stack that only allocated local storage before calling the
hash function. Instead the storage is allocated by the calling function then the hash function is
called directly. The change was believed to be trivial with no change in functionality. Although it
works for all other yespower algos it doesn't work for r8g.

I analyzed the code in detail but found nothing to explain why the hash is invalid. Getting to the root
cause would be nice but would involve extensive low level debugging.

Since the changes did not affect performance I will simply revert to the v3.12.7 code.
